### PR TITLE
Update hetzner CCM v1.15.0

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
Update hetzner CCM to 1.15.0 to gain access to ARM instances

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubermatic/kubeone/discussions/2773

**What type of PR is this?**
/kind chore

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
New Hetzner ARM instances
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

**Comment by xrstf** [comment]
All CSI related components just log

```
W0424 20:34:10.021292       1 connection.go:173] Still connecting to unix:///run/csi/socket
W0424 20:34:20.022229       1 connection.go:173] Still connecting to unix:///run/csi/socket
W0424 20:34:30.021478       1 connection.go:173] Still connecting to unix:///run/csi/socket
W0424 20:34:40.021888       1 connection.go:173] Still connecting to unix:///run/csi/socket
W0424 20:34:50.022287       1 connection.go:173] Still connecting to unix:///run/csi/socket
```

ad infinitum

File: pkg/templates/images/images.go [hunk 1]
```diff
@@ -289,10 +289,10 @@ func optionalResources() map[Resource]map[string]string {
 		DigitalOceanCSISnapshotter:               {"*": "registry.k8s.io/sig-storage/csi-snapshotter:v6.2.1"},
 
 		// Hetzner CCM
-		HetznerCCM: {"*": "docker.io/hetznercloud/hcloud-cloud-controller-manager:v1.13.2"},
+		HetznerCCM: {"*": "docker.io/hetznercloud/hcloud-cloud-controller-manager:v1.15.0"},
 
 		// Hetzner CSI
-		HetznerCSI:                   {"*": "docker.io/hetznercloud/hcloud-csi-driver:2.2.0"},
+		HetznerCSI:                   {"*": "docker.io/hetznercloud/hcloud-csi-driver:2.3.2"},
```
Comment (xmudrii) [inline pkg/templates/images/images.go:295]: This should be prefixed with `v`, see https://hub.docker.com/layers/hetznercloud/hcloud-csi-driver/v2.3.2/images/sha256-0ed4f2f51459466f994c2ce6a566819e61c6dda26c1312d4e36172b74ea97d9d?context=explore

```suggestion
		HetznerCSI:                   {"*": "docker.io/hetznercloud/hcloud-csi-driver:v2.3.2"},
```

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: pkg/templates/images/images.go (hunk 1 above)
Comment (xmudrii) [reply pkg/templates/images/images.go:295]: Additional reference (thanks @xrstf): https://github.com/hetznercloud/csi-driver/pull/407/files

**Comment by kron4eg** [comment]
@xmudrii @xrstf now it's passing!

**Review by xmudrii — APPROVED** [review]
/approve

**Pull request merged**